api/references/when-clause-contexts.md

```diff
@@ -16,37 +16,130 @@ For example, VS Code uses when clauses to enable or disable command keybindings,
+Above, the built-in **Start Debugging** command has the keyboard shortcut `kb(workbench.action.debug.start)`, which is only enabled when there is an appropriate debugger available (context key `debuggersAvailable` is true) and the editor isn't in debug mode (context key `inDebugMode` is false).
 
-## Conditional operators
+## Composing when-clauses with operators
 
-For conditional expressions, you can use the following conditional operators:
+A when-clause can consist of a single context key, e.g., `inDebugMode`, or can use various operators to express more nuanced editor state.
+
+_Note_: When-clauses, starting from VS Code version 1.77, can include parentheses. Extensions that use parentheses in when-clauses should be specified as compatible with VS Code version 1.77 and up only.
+
+### Logical operators
+
+Logical operators allow combining simple context keys or when-clause expressions that include other logical, equality, comparison, match, `in`/`not in` operators or parenthesized expressions.
 
 Operator | Symbol | Example
 -------- | ------ | -----
-Equality | `==` | `"editorLangId == typescript"`
-Inequality | `!=` | `"resourceExtname != .js"`
-Or | <code>\|\|</code> | `"isLinux`<code>\|\|</code>`isWindows"`
+Not | `!` | `"!editorReadonly"` or <code>"!(editorReadonly \|\| inDebugMode)"</code>
 And | `&&` | `"textInputFocus && !editorReadonly"`
-Not | `!` | `!editorReadonly`
-Matches | `=~` | `"resourceScheme =~ /^untitled$\|^file$/"`
-Greater than | `>` `>=` | `"gitOpenRepositoryCount >= 1"`
-Less than | `<` `<=` | `"workspaceFolderCount < 2"`
-In | `in` | `resourceFilename in supportedFolders` ([details](#in-and-not-in-conditional-operators) below)
+Or | <code>\|\|</code> | `"isLinux`<code> \|\| </code>`isWindows"`
+
+Note on logical operator precedence: the above table lists operators in order of highest to lowest precedence. Examples:
+
+Written as                             | Interpreted as
+---------------------------------------|-----------------------------------------
+`!foo && bar`                          | `(!foo) && bar`
+<code>!foo \|\| bar </code>            | `(!foo) \|\| bar`
+<code>foo \|\| bar && baz </code>      | <code>foo \|\| (bar && baz)</code>
+<code>!foo && bar \|\| baz </code>     | <code>(!foo && bar) \|\| baz</code>
+<code>!(foo \|\| bar) && baz </code>   | <code>(remains same) !(foo \|\| bar) && baz</code>
 
-### key-value when clause operator
+### Equality operators
 
-There is a key-value pair match operator for `when` clauses. The expression `key =~ value` treats the right-hand side as a regular expression to match against the left-hand side. For example, to contribute context menu items for all Docker files, one could use:
+You can check for equality a context key's value in the current editor context against a value. Note that the right-hand side is a value and not interpreted as a context key, i.e., it is not looked up in the context.
```

Comment:
Not sure this sentence reads correctly. 
Maybe "You can check for equality of a context key's value against a specified value."
Avoid Latin abbreviations so "Note that the right-hand side is a value and not interpreted as a context key, meaning it is not looked up in the context."